Python/tier2_redundancy_eliminator_cases.c.h

```diff
@@ -1267,28 +1267,32 @@
         }
 
         case _LOAD_ATTR_METHOD_WITH_VALUES: {
+            _Py_UOpsSymType *owner;
             _Py_UOpsSymType *attr;
             _Py_UOpsSymType *self = NULL;
-            attr = sym_new_unknown(ctx);
-            if (attr == NULL) goto out_of_space;
-            self = sym_new_unknown(ctx);
-            if (self == NULL) goto out_of_space;
+            owner = stack_pointer[-1];
+            PyObject *descr = (PyObject *)this_instr->operand;
+            _LOAD_ATTR_NOT_NULL_SELF
+            (void)descr;
+            (void)owner;
             stack_pointer[-1] = attr;
-            stack_pointer[0] = self;
-            stack_pointer += 1;
+            if (oparg & 1) stack_pointer[0] = self;
+            stack_pointer += (oparg & 1);
```

Comment:
I can't help feeling that the previously generated code here was more correct -- this op must only be called when the lower bit of oparg is set. I believe the generator contains some special casing to handle this -- maybe your changes hit a path where the special casing is still missing?